### ISSUE TITLE
[web]: Image.network loaded mages disappear on reload

### ISSUE THREAD

**Issue by olivierbrand** [flutter/flutter#57634]
Images pulled from the web show ok when running `flutter run -d chrome`
However when I deploy the app on Firebase using: firebase deploy and hitting the endpoint, when I refresh the page, network pulled images do not show up at all. If I force refresh on the browser, they appear again.

**Comment by olivierbrand** [comment]
Flutter doctor v:
flutter doctor -v
[✓] Flutter (Channel beta, 1.18.0-11.1.pre, on Mac OS X 10.15.5 19F53f, locale en-US)
    • Flutter version 1.18.0-11.1.pre at /Users/olivier/dev/flutter
    • Framework revision 2738a1148b (6 days ago), 2020-05-13 15:24:36 -0700
    • Engine revision ef9215ceb2
    • Dart version 2.9.0 (build 2.9.0-8.2.beta)

 
[✓] Android toolchain - develop for Android devices (Android SDK version 28.0.3)
    • Android SDK at /Users/olivier/Library/Android/sdk
    • Platform android-29, build-tools 28.0.3
    • ANDROID_HOME = /Users/olivier/Library/Android/sdk
    • Java binary at: /Applications/Android Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 1.8.0_212-release-1586-b4-5784211)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 11.4.1)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 11.4.1, Build version 11E503a
    • CocoaPods version 1.9.1

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 3.6)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin version 45.1.1
    • Dart plugin version 192.8052
    • Java version OpenJDK Runtime Environment (build 1.8.0_212-release-1586-b4-5784211)

[✓] VS Code (version 1.45.1)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.10.2

[✓] Connected device (3 available)
    • iPhone 11 Pro Max • 6CF411DE-C8BF-48E2-9116-6BFC94A510E4 • ios            • com.apple.CoreSimulator.SimRuntime.iOS-13-4 (simulator)
    • Web Server        • web-server                           • web-javascript • Flutter Tools
    • Chrome            • chrome                               • web-javascript • Google Chrome 81.0.4044.138

• No issues found!

**Comment by iapicca** [comment]
Hi @olivierbrand 
can you please provide  a reproducible minimal code sample.
Thank you

**Comment by olivierbrand** [comment]
Below is the entire main.dart file. I have changed it so it just use hardcoded data as you can see.
I am deploying to firebase hosting
The current code below is executed on the following URL: https://nextread-6c2fa.web.app/#/
First load, always works, if you reload the page, you will see the image disapear.
If you force reload the page, images are there again

`import 'dart:async';

import 'package:flutter/material.dart';

void main() {
  runApp(MyApp());
}

class MyApp extends StatelessWidget {
  // This widget is the root of your application.
  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      title: 'NextRead',
      theme: ThemeData(
        // This is the theme of your application.
        //
        // Try running your application with "flutter run". You'll see the
        // application has a blue toolbar. Then, without quitting the app, try
        // changing the primarySwatch below to Colors.green and then invoke
        // "hot reload" (press "r" in the console where you ran "flutter run",
        // or simply save your changes to "hot reload" in a Flutter IDE).
        // Notice that the counter didn't reset back to zero; the application
        // is not restarted.
        primarySwatch: Colors.blue,
        // This makes the visual density adapt to the platform that you run
        // the app on. For desktop platforms, the controls will be smaller and
        // closer together (more dense) than on mobile platforms.
        visualDensity: VisualDensity.adaptivePlatformDensity,
      ),
      home: MyHomePage(title: 'NextRead'),
    );
  }
}

class MyHomePage extends StatefulWidget {
  MyHomePage({Key key, this.title}) : super(key: key);

  // This widget is the home page of your application. It is stateful, meaning
  // that it has a State object (defined below) that contains fields that affect
  // how it looks.

  // This class is the configuration for the state. It holds the values (in this
  // case the title) provided by the parent (in this case the App widget) and
  // used by the build method of the State. Fields in a Widget subclass are
  // always marked "final".

  final String title;

  @override
  _MyHomePageState createState() => _MyHomePageState();
}

class _MyHomePageState extends State<MyHomePage> {
  void initState() {
    super.initState();
  }

  void dispose() {
    super.dispose();
  }

  Future<List<BookInfo>> getBookInfo() {
    List<BookInfo> books = [
      BookInfo('book1',
          'http://books.google.com/books/content?id=LEBivgAACAAJ&printsec=frontcover&img=1&zoom=1&source=gbs_api'),
      BookInfo('book2',
          'http://books.google.com/books/content?id=mWuvowui4r8C&printsec=frontcover&img=1&zoom=1&edge=curl&source=gbs_api')
    ];

    return Future<List<BookInfo>>.value(books);
  }

  @override
  Widget build(BuildContext context) {
    // This method is rerun every time setState is called, for instance as done
    // by the _incrementCounter method above.
    //
    // The Flutter framework has been optimized to make rerunning build methods
    // fast, so that you can just rebuild anything that needs updating rather
    // than having to individually change instances of widgets.
    return Scaffold(
      appBar: AppBar(
        // Here we take the value from the MyHomePage object that was created by
        // the App.build method, and use it to set our appbar title.
        title: Text(widget.title),
      ),
      body: Center(
        // Center is a layout widget. It takes a single child and positions it
        // in the middle of the parent.
        child: Column(
          // Column is also a layout widget. It takes a list of children and
          // arranges them vertically. By default, it sizes itself to fit its
          // children horizontally, and tries to be as tall as its parent.
          //
          // Invoke "debug painting" (press "p" in the console, choose the
          // "Toggle Debug Paint" action from the Flutter Inspector in Android
          // Studio, or the "Toggle Debug Paint" command in Visual Studio Code)
          // to see the wireframe for each widget.
          //
          // Column has various properties to control how it sizes itself and
          // how it positions its children. Here we use mainAxisAlignment to
          // center the children vertically; the main axis here is the vertical
          // axis because Columns are vertical (the cross axis would be
          // horizontal).
          mainAxisAlignment: MainAxisAlignment.center,
          children: <Widget>[
            Flexible(
              child: FutureBuilder(
                future: getBookInfo(),
                builder: (context, snapshot) {
                  if (!snapshot.hasData) {
                    return Container();
                  } else {
                    return ListView.separated(
                      padding: const EdgeInsets.all(16),
                      separatorBuilder: (BuildContext context, int index) =>
                          const Padding(
                              padding: EdgeInsets.all(8),
                              child: Divider(
                                thickness: 2,
                                color: Colors.black12,
                              )),
                      itemCount: snapshot.data.length,
                      itemBuilder: (context, index) {
                        BookInfo book = snapshot.data[index];
                        return ListTile(
                            leading: Image.network(
                              book.thumbnail,
                              fit: BoxFit.cover,
                            ),
                            title: Text(book.title));
                      },
                    );
                  }
                },
              ),
            ),
          ],
        ),
      ),
    );
  }
}

class BookInfo {
  String title;
  String thumbnail;

  BookInfo(this.title, this.thumbnail);
}
`

**Comment by jonahwilliams** [comment]
If you're serving resources over http you can't use the service worker. I'm not sure where we should document this, but removing the service worker script tag from your index.html will "fix it"

**Comment by olivierbrand** [comment]
Could you explain what this service worker does and if not working, why is it being generated? Or is it Firebase that added it? Thanks for a detailed explanation and any articles that would cover this.

**Comment by olivierbrand** [comment]
So maybe an issue with Firebase Hosting, replacing http: with https: within my dart code seems to have fixed the issue. Strange as it worked all the time until I reload the page. A force reload was also fixing the issue but temporarily. Service worker seems to be fine

**Comment by yjbanov** [comment]
I changed the description. What we should do is not start the service worker on unsupported URLs (http is only supported for `localhost`).

**Comment by jonahwilliams** [comment]
The service worker won't start on unsupported domains. The issue with this specific configuration is that the main bundle is served via https but the assets are served under http.

**Comment by olivierbrand** [comment]
Yes service worker does not seem to be involved. It is also strange theta Google API I am using returns assets under http and not https.... Is there a way to have a mix of http and https via config? For now I am translating incoming assets served by http by Google Book API into https before they get loaded by the Flutter web app, that solves the problem but is a hack

**Comment by jonahwilliams** [comment]
No, if you're using http at all you simply cannot use a service worker

**Comment by olivierbrand** [comment]
I need to triple check, maybe a Firebase Hosting replication issue because when I removed the service worker script from index.html, I had the same issues. And the logs I provided earlier show that the service worker was failing on http assets even though the service worker script was disabled??
